spec/resolver-fixtures/invalid-absolute-grant-error.yml

```diff
@@ -0,0 +1,9 @@
+policy: |
+  - !group root-owner
+  - !policy
+    id: myapp
+    body:
+    - !grant
+      role: !group /root-owner
+      member: !user
+error: "Illegal absolute id: /root-owner"
```

Comment:
Do we need all three of the "invalid-absolute-*" tests? Aren't they all testing the same error condition?

Reply:
Kind of, but not entirely: the absolute ids are in different places. But I did hesitate to include all of them, since it's not complete anyway: I'd have to include one for each of the `subject_id` implementations, and that felt like too much of the same thing. What do you think?

Reply:
Well, it sort of feels like there could be two sets of tests here. One set would confirm that each resolver produces the correct result. Another set would confirm that policy traversal works. With those two in place, you could get rid of `AbsoluteChecker`, and its requirement that the types implement `subject_id`.

For this PR, I guess I'd be satisfied with a few specs that demonstrate that `PolicyNamespaceResolver#prepend_namespace` produces the correct results. How much testing of policy traversal, and whether or not to keep `AbsoluteChecker`, I'll leave up to you.

Reply:
I'm not sure I understand. `AbsoluteChecker` is there to raise the error if there are any ids left absolute; this doesn't have much to do with `PolicyNamespaceResolver`, which doesn't touch absolute references. Then `RelativePathResolver` replaces them in legal places. There still needs to be something to raise an error if they are somewhere in an illegal place. And I didn't change how policy traversal works at all.